RocketChat/Rocket.Chat
# fix: Surface elements (Modals, Contextual Bars) fail to respond when triggered by app submit events

## Proposed changes (including videos or screenshots)


**Purpose:**
This PR addresses a specific issue in the app's UI behavior, particularly in the modal and contextual bar interfaces used for updates.

**Problem:**
Previously, the update modal/contextual bar did not consistently close upon the submission of updates, leading to potential confusion and a disjointed user experience.

**Solution:**
To resolve this, the PR introduces a refinement in the logic controlling the modal/contextual bar. It now retrieves the interaction type involved in the update process. This enhancement allows the system to accurately determine if and when the modal/contextual bar should close after a submission event.

## Issue(s)


- https://rocketchat.atlassian.net/browse/CORE-138
- https://rocketchat.atlassian.net/browse/SUP-459

## Steps to test or reproduce


1. Install (or create one) an app  that updates a modal/contextual
2. Trigger the action that would update the surface
3. It should update the surface instead of just closing it

## Further comments


Here is an app to test this issue:

[test-surface-changes_0.0.1.zip](https://github.com/RocketChat/Rocket.Chat/files/14932874/test-surface-changes_0.0.1.zip)
[test-surface-changes_0.0.1-not-compiled.zip](https://github.com/RocketChat/Rocket.Chat/files/14932889/test-surface-changes_0.0.1-not-compiled.zip)
